internal/controller/postgrescluster/cluster.go

```diff
@@ -296,6 +298,9 @@ func (r *Reconciler) reconcileDataSource(ctx context.Context,
 		// restore ID for bootstrap
 		restoreID = "~pgo-bootstrap-" + cluster.GetName()
 		dataSource = cluster.Spec.DataSource.PostgresCluster
+		if dataSource == nil {
```

Comment:
:thinking: If both a `PostgresCluster` data source and a `PGBackRest` data source are both defined, the `PostgresCluster` data source appears take precedence, if I understand correctly? I did see the note in pkg/apis/postgres-operator.crunchydata.com/v1beta1/postgrescluster_types.go, i.e.
```
The PGBackRest field is incompatible with the PostgresCluster field: only one
data source can be used for pre-populating a new PostgreSQL cluster
```
but I was curious how/if we're handling the 'both' case.

Reply:
You're right that the `PostgresCluster` would take precedence (i.e., all the switch statements test first for that); I was thinking of a future webhook validation that would only allow one or the other, but I should probably call it out more in the documentation. (Again: possibly not a problem if we combine the datasource objects into one object with an enum field.)

Reply:
Added.